controllers/argocd/configmap.go

```diff
@@ -331,9 +360,27 @@ func (r *ReconcileArgoCD) reconcileArgoConfigMap(cr *argoprojv1a1.ArgoCD) error
 	}
 
 	cm.Data[common.ArgoCDKeyOIDCConfig] = getOIDCConfig(cr)
+
+	// new format with subkeys for ResourceCustomizations
+	if c := getNewResourceCustomizations(cr); c != nil {
+		for k, v := range c {
+			cm.Data[k] = v
+		}
+	}
+
+	// old format of ResourceCustomizations, use should not be encouraged but is supported for now
 	if c := getResourceCustomizations(cr); c != "" {
+		// Emit event providing users with deprecation notice for ResourceCustomization
+		if !ResourceCustomizationDeprecationWarningEmitted {
+			err := argoutil.CreateEvent(r.Client, "Warning", "Deprecated", "ResourceCustomizations is deprecated, please use the new format, ResourceCustomizationsNew, instead.", "DeprecationNotice", cr.ObjectMeta, cr.TypeMeta)
```

Comment:
If we have a timeline for when we will remove support for the old format we should mention it here